### PR TITLE
Ensure stack size is computed for stdcall name mangling

### LINKED ISSUE #46667: Thousands failures failed with ndirect.m_cbStackArgumentSize != 0xFFFF
Thousands failures failed in job: [runtime-coreclr crossgen2-composite 20210105.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=939723&view=ms.vss-test-web.build-test-results-tab&runId=29698044&resultId=100145&paneView=debug)

Error message
~~~
Could Not Find C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\composite-r2r.dll.rsp

Assert failure(PID 4948 [0x00001354], Thread: 7536 [0x1d70]): ndirect.m_cbStackArgumentSize != 0xFFFF

CORECLR! NDirectMethodDesc::GetStackArgumentSize + 0x27 (0x709deb63)
CORECLR! NDirectMethodDesc::FindEntryPointWithMangling + 0xEE (0x709db1f2)
CORECLR! NDirectMethodDesc::FindEntryPointWithSuffix + 0x50 (0x709db285)
CORECLR! NDirectMethodDesc::FindEntryPoint + 0xD6 (0x709db0d3)
CORECLR! NDirect::NDirectGetEntryPoint + 0x26E (0x7093171b)
CORECLR! NDirect::NDirectLink + 0x251 (0x70931a58)
CORECLR! NDirectImportWorker + 0x1E7 (0x70934440)
CORECLR! NDirectImportThunk + 0x8 (0x70d77c33)
FRAMEWORK-R2R! <no symbol> + 0x0 (0x6db9c3e4)
FRAMEWORK-R2R! <no symbol> + 0x0 (0x6db9f54f)
    File: F:\workspace\_work\1\s\src\coreclr\vm\method.hpp Line: 3255
    Image: C:\h\w\A64408D3\p\CoreRun.exe


Return code:      1
Raw output file:      C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Reports\JIT.IL_Conformance\Old\Conformance_Base\add_ovf_i8\add_ovf_i8.output.txt
Raw output:
BEGIN EXECUTION
add_ovf_i8.dll
        1 file(s) copied.
Response file: C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\\composite-r2r.dll.rsp
C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\IL-CG2\.dll
-o:C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\\composite-r2r.dll
--targetarch:x86
--verify-type-and-field-layout
-O
-r:C:\h\w\A64408D3\p\System..dll
-r:C:\h\w\A64408D3\p\Microsoft.*.dll
-r:C:\h\w\A64408D3\p\mscorlib.dll
-r:C:\h\w\A64408D3\p\netstandard.dll
" "dotnet" "C:\h\w\A64408D3\p\crossgen2\crossgen2.dll" @"C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\\composite-r2r.dll.rsp"   --composite"
Emitting R2R PE file: C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\\composite-r2r.dll
Emitting R2R PE file: C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\add_ovf_i8.dll
 "C:\h\w\A64408D3\p\corerun.exe" add_ovf_i8.dll 
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\A64408D3\p
> C:\h\w\A64408D3\w\A5FB0904\e\JIT\IL_Conformance\Old\Conformance_Base\add_ovf_i8\add_ovf_i8.cmd
Expected: True
Actual:   False
~~~

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #46667

### CODE REVIEW

**Comment by jkotas** [comment]
triggered the crossgen2-composite job to see whether it is fixing the problem

**Comment by jkotas** [comment]
This needs more work...

File: src/coreclr/vm/dllimport.h [hunk 1]
```diff
@@ -211,6 +211,7 @@ inline bool SF_IsNGENedStubForProfiling(DWORD dwStubFlags) { LIMITED_METHOD_CONT
 inline bool SF_IsDebuggableStub        (DWORD dwStubFlags) { LIMITED_METHOD_CONTRACT; return (dwStubFlags < NDIRECTSTUB_FL_INVALID && 0 != (dwStubFlags & NDIRECTSTUB_FL_GENERATEDEBUGGABLEIL)); }
 inline bool SF_IsCALLIStub             (DWORD dwStubFlags) { LIMITED_METHOD_CONTRACT; return (dwStubFlags < NDIRECTSTUB_FL_INVALID && 0 != (dwStubFlags & NDIRECTSTUB_FL_UNMANAGED_CALLI)); }
 inline bool SF_IsStubWithCctorTrigger  (DWORD dwStubFlags) { LIMITED_METHOD_CONTRACT; return (dwStubFlags < NDIRECTSTUB_FL_INVALID && 0 != (dwStubFlags & NDIRECTSTUB_FL_TRIGGERCCTOR)); }
+inline bool SF_IsForNumParamBytes      (DWORD dwStubFlags) { LIMITED_METHOD_CONTRACT; return (dwStubFlags < NDIRECTSTUB_FL_INVALID && 0 != (dwStubFlags & NDIRECTSTUB_FL_FOR_NUMPARAMBYTES)); }
```
Comment (jkotas) [inline src/coreclr/vm/dllimport.h:214]: Partial undo of https://github.com/dotnet/runtime/commit/d8715a87f785f69af062139bb37788d433bb980e#diff-f50a5caf825717b9c948187c6e4f1e976267c4cb311ec05e37858c10365032f0L213

**Comment by jkotas** [comment]
The remaining test failures are known issues.

**Review by trylek — APPROVED** [review]
LGTM, thank you!

**Pull request merged**